src/wp-includes/formatting.php

```diff
@@ -5394,6 +5394,28 @@ function normalize_whitespace( $str ) {
  * @return string The processed string.
  */
 function wp_strip_all_tags( $text, $remove_breaks = false ) {
+	if ( is_null( $text ) ) {
+		return '';
+	}
+
+	if ( ! is_string( $text ) ) {
+		if ( WP_DEBUG ) {
+			trigger_error(
+				sprintf(
+					/* translators: 1: The function name, 2: The argument number, 3: The expected type, 4: The provided type. */
+					__( 'Warning: %1$s expects parameter %2$d to be a %3$s, %4$s given.' ),
+					__FUNCTION__,
+					'1',
+					'string',
+					gettype( $text )
+				),
+				E_USER_WARNING
+			);
+		}
+
+		return '';
+	}
+
```

Comment:
```suggestion

	$text = (string) $text;
```

Future proofing (hopefully)

Reply:
Added in [9862146](https://github.com/WordPress/wordpress-develop/pull/3132/commits/986214639da8dff312db5ade06e5a79e64f8cac6).

Reply:
PHP still juggles scalars and a scalar check is already being done. This is overhead which may potentially _hide_ future issues. IMO let's handle future issues in the future and not add code which is _currently_ redundant.

Reply:
@jrfnl works for me. Sorry Colin, are you able to revert 986214639da8dff312db5ade06e5a79e64f8cac6

Reply:
that should read: revert the relevant portion of....